kubernetes/kiali_monitoring_client.go

```diff
@@ -37,19 +40,32 @@ func NewKialiMonitoringClient() (*KialiMonitoringClient, error) {
 		return nil, err
 	}
 
-	client, err := newClientForAPI(config, kialiMonitoringGroupVersion, types)
+	clientV1Alpha1, err := newClientForAPI(config, v1alpha1.KialiMonitoringGroupVersion, types)
+	if err != nil {
+		return nil, err
+	}
+	clientV1Alpha2, err := newClientForAPI(config, v1alpha2.KialiMonitoringGroupVersion, types)
+	if err != nil {
+		return nil, err
+	}
 	return &KialiMonitoringClient{
-		client: client,
+		clientV1Alpha1: clientV1Alpha1,
+		clientV1Alpha2: clientV1Alpha2,
 	}, err
 }
 
 // GetDashboard returns a MonitoringDashboard for the given name
-func (in *KialiMonitoringClient) GetDashboard(namespace string, name string) (*MonitoringDashboard, error) {
-	result, err := in.client.Get().Namespace(namespace).Resource("monitoringdashboards").SubResource(name).Do().Raw()
-	if err != nil {
-		return nil, err
+func (in *KialiMonitoringClient) GetDashboard(namespace, name string) (*v1alpha2.MonitoringDashboard, error) {
+	result := v1alpha2.MonitoringDashboard{}
+	err := in.clientV1Alpha2.Get().Namespace(namespace).Resource("monitoringdashboards").SubResource(name).Do().Into(&result)
+	if errors.IsNotFound(err) {
+		// Try with older version
```

Comment:
@lucasponce if you have any suggestion here to do better, I'm all ears. Basically, to allow backward compatibility, I'm trying to get dashboards first with client v1alpha2, if not found I retry with client for v1alpha1.
But when the "monitoringdashboards" CDR is installed on a cluster, I think it's only for 1 version (so we cannot have at the same time different versions of resources, afaik). So maybe there's a way to check what CRD is installed on the cluster to use directly the right client? Any idea?